server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java

```diff
@@ -87,13 +87,13 @@ public void handle(Request request, Response response) {
 
   private static void ensureNoneIsMarkedAsDefault(QualityProfileDto profile, List<QualityProfileDto> descendants) {
     if (profile.isDefault()) {
-      throw new IllegalStateException("Profile is marked as 'default' and cannot be deleted");
+      throw new IllegalArgumentException("Profile is marked as 'default' and cannot be deleted");
```

Comment:
The texts will be

> Profile 'profile-name' cannot be deleted because it is marked as default


and

> Profile 'profile-name' cannot be deleted because its descendant named 'profile-name' is marked as default